gallery_dl/extractor/fanbox.py

```diff
@@ -0,0 +1,166 @@
+            "title": pbody["title"],
+            "fee_required": pbody["feeRequired"],
+            "published": pbody["publishedDatetime"],
+            "updated": pbody["updatedDatetime"],
+            "type": pbody["type"],
+            "tags": pbody["tags"],
+            "creator_id": pbody["creatorId"],
+            "has_adult_content": pbody["hasAdultContent"],
+            "post_url": self.root+"/@"+pbody["creatorId"]+"/posts/"+post_id,
+            "creator_user_id": pbody["user"]["userId"],
+            "creator_name": pbody["user"]["name"],
+            "text": pbody["body"].get("text", None) if pbody["body"] else None,
+            "is_cover_image": False
+        }
+
+        num = 0
+        if "coverImageUrl" in pbody and pbody["coverImageUrl"]:
+            final_post = dict(post)
+            final_post["isCoverImage"] = True
+            final_post["file_url"] = pbody["coverImageUrl"]
+            final_post = text.nameext_from_url(
+                pbody["coverImageUrl"], final_post
+            )
+            final_post["num"] = num
+            num += 1
+            yield pbody["coverImageUrl"], final_post
+
+        for group in ["images", "imageMap"]:
+            if group in (pbody["body"] or []):
+                for item in pbody["body"][group]:
+                    final_post = dict(post)
+                    final_post["file_url"] = item["originalUrl"]
+                    final_post = text.nameext_from_url(
+                        item["originalUrl"], final_post
+                    )
+                    if "extension" in item:
+                        final_post["extension"] = item["extension"]
+                    final_post["file_id"] = item.get("id", None)
+                    final_post["width"] = item.get("width", None)
+                    final_post["height"] = item.get("height", None)
+                    final_post["num"] = num
+                    num += 1
+                    yield item["originalUrl"], final_post
+
+        for group in ["files", "fileMap"]:
+            if group in (pbody["body"] or []):
+                for item in pbody["body"][group]:
+                    final_post = dict(post)
+                    final_post["file_url"] = item["url"]
+                    final_post = text.nameext_from_url(item["url"], final_post)
+                    if "extension" in item:
+                        final_post["extension"] = item["extension"]
+                    if "name" in item:
+                        final_post["filename"] = item["name"]
+                    final_post["file_id"] = item.get("id", None)
+                    final_post["num"] = num
+                    num += 1
+                    yield item["url"], final_post
+
+
+class FanboxCreatorExtractor(FanboxExtractor):
+    """Extractor for a fanbox creator's works"""
+    subcategory = "creator"
+    pattern = (r"(?:https?://)?([a-zA-Z0-9_-]+)\.fanbox\.cc/?$|"
+               r"(?:https?://)?(?:www\.)?fanbox\.cc/@([^/?#]+)/?$")
```

Comment:
I think it'd be better to use a `BASE_PATTERN` like [here](https://github.com/mikf/gallery-dl/blob/78d7ee3ef4eed38b24591fcd553bb3a08e3215db/gallery_dl/extractor/deviantart.py#L21-L25) and [here](https://github.com/mikf/gallery-dl/blob/78d7ee3ef4eed38b24591fcd553bb3a08e3215db/gallery_dl/extractor/deviantart.py#L373)